dask/array/core.py

```diff
@@ -1383,6 +1383,44 @@ def vindex(self):
         """
         return IndexCallable(self._vindex)
 
+    def _blocks(self, key):
+        from .slicing import normalize_index
+        if not isinstance(key, tuple):
+            key = (key,)
+        key = tuple([k] if isinstance(k, Number) else k for k in key)
```

Comment:
We should consider using `slice(k, k+1)` here instead of `[k]`. I'm not entirely sure which is better, yet, but these can differ due to [weird edge cases of fancy indexing](http://www.numpy.org/neps/nep-0021-advanced-indexing.html).

Reply:
OK, let's definitely switch to `slice(k, k+1)` here.

Consider the case of indexing a 3D array like `array.blocks[0, :, 0]` with `chunks=1`. Now compare what the result would look like:
```
>>> x = np.zeros((5, 6, 7))
>>> x[:1, :, :1].shape  # seems reasonable
(1, 6, 1)
>>> x[[0], :, [0]].shape  # where did the last dimension go?
(1, 6)
```

I could go on to more examples of strange behavior, but basically we want to stay as close to "basic indexing" as possible, rather than inadvertently triggering "advanced indexing".

It would be good to note this explicitly in the docs, though (that integers get converted into slices), so users know how to get vectorized indexing if/when desired by replacing integers with lists/arrays.